hedera-node/src/main/java/com/hedera/services/store/contracts/SizeLimitedStorage.java

```diff
@@ -353,9 +353,11 @@ private void commitPendingUpdates() {
 			var firstKey = newFirstKeys.containsKey(id) ? newFirstKeys.get(id) : firstKeyLookup(id);
 			for (final var changedKey : changeSet) {
 				final var newValue = newMappings.get(changedKey);
+				final var preInsertSize = curStorage.size();
 				firstKey = storageUpserter.upsertMapping(changedKey, newValue, firstKey, firstValue, curStorage);
-				firstValue = firstKey.equals(changedKey) ? newValue : null;
-			}
+				// If newValue was just added to the map, it is the mutable root value; but if we only
+				// updated the existing root value, then newValue is NOT the mutable root value
+				firstValue = (firstKey.equals(changedKey) && curStorage.size() > preInsertSize) ? newValue : null;			}
```

Comment:
Formatting needs to be fixed.  I can't put hard tabs into suggestions (another piece of evidence that spaces are better than tabs)